samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,203 @@
+// Deep learning based Object Detection and Instance Segmentation using Mask R-CNN
+// VIDEO 360 DEMO: https://www.youtube.com/watch?v=0tU8991QgE8
+// Downloads:
+// http://download.tensorflow.org/models/object_detection/mask_rcnn_inception_v2_coco_2018_01_28.tar.gz
+// https://raw.githubusercontent.com/opencv/opencv_extra/master/testdata/dnn/mask_rcnn_inception_v2_coco_2018_01_28.pbtxt
+// https://raw.githubusercontent.com/spmallick/learnopencv/master/Mask-RCNN/mscoco_labels.names
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <string>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+using namespace cv;
+using namespace dnn;
+using namespace std;
+
+string keys =
+    "{ help  h     | | Print help message. \nUsage \n\t\t./mask_rcnn --image=logo.jpg \n\t\t ./mask_rcnn --video=teste.mp4}"
+    "{ image m     |<none>| Path to input image file.  }"
+    "{ video v     |<none>| Path to input video file.  }"
+    "{ classes n   |./mscoco_labels.names | Path to a text file with names of classes.  }"
+    "{ model w     |./mask_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb | The pre-trained weights.  }"
+    "{ height H     |800|Preprocess input image by resizing to a specific height }"
+    "{ width W     |800|Preprocess input image by resizing to a specific width }"
+    "{ config c    |./mask_rcnn_inception_v2_coco_2018_01_28.pbtxt | iThe text graph file that has been tuned by the OpenCV’s DNN support group  }"
+    "{ cthr        | .5 | Confidence threshold. }"
+    "{ mthr        | .4 | Mask threshold. }";
+
+void drawBox(Mat& frame, int classId, float conf, Rect box, Mat& objectMask, float maskThreshold, vector<Scalar> colors, vector<string> classes);
+
+void postprocess(Mat& frame, const vector<Mat>& outs, float confThreshold, float maskThreshold, vector<Scalar> colors, vector<string> classes);
```

Comment:
Helper local functions should be "static".

Reply:
done